torch/_higher_order_ops/cond.py

```diff
@@ -321,29 +322,7 @@ def _same_meta_except_requires_grad(true_out, false_out):
         "call_function", func_overload, proxy_args, {}
     )
 
-    # At this point, we're *guaranteed* that whether an output came from the
-    # true or false branch is indistinguishable. So, as this is just for tracing
-    # purposes, choose the true branch.
-
-    # TODO: the unbacked symbol allocations MUST NOT leak out, if you want to
-    # support this we need to arrange for the reenter_make_fx unbacked SymInts
-    # to be used, AND we need to arrange for some sort of unification between
-    # the two branches (but not really unification; e.g., if one branch
-    # returns [u0] and the other returns [5] this is OK but you MUST NOT
-    # conclude the result is 5.  Also if one branch returns [3] and another
-    # branch returns [5] you can make it work by immediately allocating a new
-    # unbacked SymInt here).
-    ignore_fresh_unbacked = contextlib.nullcontext()
-    if (fake_mode := detect_fake_mode()) and fake_mode.shape_env:
-        ignore_fresh_unbacked = fake_mode.shape_env.ignore_fresh_unbacked_symbols()
-
-    # TODO: Uhh.... it shouldn't matter, but changing this to true_fn results in
-    # a FakeTensorMode error :
-    # `Current active mode <class 'torch._subclasses.fake_tensor.FakeTensorMode'> not registered`
-    # TODO Sometimes the operands are not completely FakeTensor, something seems went wrong in
-    # dynamo? Because of that it runs real computation sometimes and re-triggering downstream dispatch keys.
-    with ignore_fresh_unbacked:
-        out = false_fn(*operands)
+    out = func_overload(pred, true_fn, false_fn, operands)
```

Comment:
This is worth doing on its own as a PR, and figuring out if it works or doesn't work